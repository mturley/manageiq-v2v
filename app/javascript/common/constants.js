export const MAX_LENGTH_NAMES = 24;
export const MAX_LENGTH_DESCRIPTIONS = 128;

const maxLengthIs = __('Maximum length is %s characters.');
const requiredMaxLengthIs = __('Required. Maximum length is %s characters.');
const youHaveReachedMax = __('You have reached the maximum length of %s characters.');

export const validation = {
  name: {
    help: sprintf(maxLengthIs, MAX_LENGTH_NAMES),
    requiredMessage: sprintf(requiredMaxLengthIs, MAX_LENGTH_NAMES),
    maxLength: MAX_LENGTH_NAMES,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_NAMES)
  },
  description: {
    help: sprintf(maxLengthIs, MAX_LENGTH_DESCRIPTIONS),
    maxLength: MAX_LENGTH_DESCRIPTIONS,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_DESCRIPTIONS)
  }
};

export const OPENSTACK = 'openstack';
export const RHV = 'rhevm';

export const V2V_TARGET_PROVIDERS = [
  {
    name: __('Red Hat Virtualization'),
    id: RHV,
    type: 'ManageIQ::Providers::Redhat::InfraManager'
  },
  {
    name: __('Red Hat OpenStack Platform'),
    id: OPENSTACK,
    type: 'ManageIQ::Providers::Openstack::CloudManager'
  }
];

export const CONVERSION_HOST_TYPES = {
  [RHV]: 'ManageIQ::Providers::Redhat::InfraManager::Host',
  [OPENSTACK]: 'ManageIQ::Providers::Openstack::CloudManager::Vm'
};

export const TRANSFORMATION_MAPPING_ITEM_SOURCE_TYPES = {
  cluster: 'EmsCluster',
  datastore: 'Storage',
  network: 'Lan'
};

export const TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES = {
  [OPENSTACK]: {
    cluster: 'CloudTenant',
    datastore: 'CloudVolumeType',
    network: 'CloudNetwork'
  },
  [RHV]: {
    cluster: 'EmsCluster',
    datastore: 'Storage',
    network: 'Lan'
  }
};
