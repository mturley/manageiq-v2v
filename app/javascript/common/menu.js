import http from './http';

const updateUrl = '/migration/menu_section_url?section=migration&url=/migration%23';
const activeMenuItemSelector = '[data-target="#menu-migration"] > div > ul > li[class="list-group-item active"]';
const plansMenuItemId = 'menu_item_plans';
const mappingsMenuItemId = 'menu_item_mappings';
const settingsMenuItemId = 'menu_item_settings';
const pathToMenuItem = {
  plan: plansMenuItemId,
  plans: plansMenuItemId,
  mappings: mappingsMenuItemId,
  settings: settingsMenuItemId
};

export const updateVerticalMenu = path => {
  // POST /migration/menu_section_url to update session[:tab_url] on server
  http.post(updateUrl + path);

  // Remove 'active' class from current active menu item
  const activeMenuItem = document.querySelector(activeMenuItemSelector);
  if (activeMenuItem !== null) {
    activeMenuItem.classList.remove('active');
  }

  // Based on location.pathname, set menu item id
  const menu_item_id = pathToMenuItem[path.split('/')[1]];

  // Add 'active' class to menu_item_id
  const newActiveMenuItem = document.querySelector(`[id="${menu_item_id}"]`);
  if (newActiveMenuItem !== null) {
    newActiveMenuItem.classList.add('active');
  }
};
