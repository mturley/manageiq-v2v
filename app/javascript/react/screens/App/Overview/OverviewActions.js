import URI from 'urijs';
import API from '../../../../common/API';
import { V2V_NOTIFICATION_ADD } from '../common/NotificationList/NotificationConstants';
import { formatDateTime } from '../../../../components/dates/MomentDate';

import {
  ARCHIVE_TRANSFORMATION_PLAN,
  DELETE_TRANSFORMATION_PLAN,
  CREATE_V2V_TRANSFORMATION_PLAN_REQUEST,
  DELETE_INFRASTRUCTURE_MAPPING,
  FETCH_DATASTORES,
  FETCH_NETWORKS,
  FETCH_PROVIDERS,
  FETCH_PROVIDERS_URL,
  FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS,
  FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS,
  FETCH_V2V_ALL_REQUESTS_WITH_TASKS,
  FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS,
  FETCH_V2V_TRANSFORMATION_MAPPINGS,
  FETCH_V2V_TRANSFORMATION_PLANS,
  HIDE_CONFIRM_MODAL,
  HIDE_DELETE_CONFIRMATION_MODAL,
  HIDE_MAPPING_WIZARD,
  SET_MAPPING_TO_DELETE,
  SHOW_CONFIRM_MODAL,
  SHOW_DELETE_CONFIRMATION_MODAL,
  SHOW_MAPPING_WIZARD,
  SHOW_PLAN_WIZARD,
  V2V_FETCH_CLUSTERS,
  V2V_RETRY_MIGRATION,
  V2V_SCHEDULE_MIGRATION,
  V2V_SET_MIGRATIONS_FILTER,
  V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL,
  YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL,
  SHOW_PLAN_WIZARD_EDIT_MODE
} from './OverviewConstants';

export const showConfirmModalAction = modalOptions => ({
  type: SHOW_CONFIRM_MODAL,
  payload: modalOptions
});

export const hideConfirmModalAction = () => ({
  type: HIDE_CONFIRM_MODAL
});

export const showMappingWizardAction = () => dispatch => {
  dispatch({
    type: SHOW_MAPPING_WIZARD
  });
};

export const showPlanWizardAction = id => dispatch => {
  dispatch({
    type: SHOW_PLAN_WIZARD,
    payload: id
  });
};

export const showPlanWizardEditModeAction = id => dispatch => {
  dispatch({
    type: SHOW_PLAN_WIZARD_EDIT_MODE,
    editingPlanId: id
  });
};

export const fetchProvidersAction = () => dispatch => {
  dispatch({
    type: FETCH_PROVIDERS,
    payload: API.get(FETCH_PROVIDERS_URL)
  });
};

const _createTransformationPlanRequestActionCreator = url => dispatch =>
  dispatch({
    type: CREATE_V2V_TRANSFORMATION_PLAN_REQUEST,
    payload: {
      promise: API.post(url, { action: 'order' }),
      data: url
    }
  });

export const createTransformationPlanRequestAction = url => {
  const uri = new URI(url);
  return _createTransformationPlanRequestActionCreator(uri.toString());
};

const _getTransformationMappingsActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_TRANSFORMATION_MAPPINGS,
    payload: API.get(url)
  });

export const fetchTransformationMappingsAction = url => {
  const uri = new URI(url);
  return _getTransformationMappingsActionCreator(uri.toString());
};

const fetchTasksForAllRequests = (allRequests, archived, dispatch) => {
  if (allRequests.length > 0) {
    dispatch({
      type: archived ? FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS : FETCH_V2V_ALL_REQUESTS_WITH_TASKS,
      payload: new Promise((resolve, reject) => {
        API.post('/api/requests?expand=resource&attributes=miq_request_tasks,service_template', {
          action: 'query',
          resources: allRequests
        })
          .then(responseRequestsWithTasks => {
            resolve(responseRequestsWithTasks);
          })
          .catch(e => reject(e));
      })
    });
  }
};

const collectAllRequests = plan => plan.miq_requests.map(request => Object.assign({}, { href: request.href }));

const _getTransformationPlansActionCreator = (url, archived) => dispatch =>
  dispatch({
    type: archived ? FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS : FETCH_V2V_TRANSFORMATION_PLANS,
    payload: new Promise((resolve, reject) => {
      API.get(url)
        .then(response => {
          resolve(response);
          const allPlansWithRequests = response.data.resources;

          const allRequests = [];
          const mergedRequests = [].concat(
            ...allRequests.concat(allPlansWithRequests.map(plan => collectAllRequests(plan)))
          );

          fetchTasksForAllRequests(mergedRequests, archived, dispatch);
        })
        .catch(e => reject(e));
    })
  });

export const fetchTransformationPlansAction = ({ url, archived }) => {
  const uri = new URI(url);
  return _getTransformationPlansActionCreator(uri.toString(), archived);
};

const _getServiceTemplateAnsiblePlaybooksActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS,
    payload: API.get(url)
  });

export const fetchServiceTemplateAnsiblePlaybooksAction = url => {
  const uri = new URI(url);
  return _getServiceTemplateAnsiblePlaybooksActionCreator(uri.toString());
};

export const continueToPlanAction = id => dispatch => {
  dispatch({
    type: HIDE_MAPPING_WIZARD
  });
  dispatch({
    type: SHOW_PLAN_WIZARD,
    payload: { id }
  });
};

const _getClustersActionCreator = url => dispatch =>
  dispatch({
    type: `${V2V_FETCH_CLUSTERS}`,
    payload: API.get(url)
  });

export const fetchClustersAction = url => {
  const uri = new URI(url);
  return _getClustersActionCreator(uri.toString());
};

const _getDatastoresActionCreator = url => dispatch =>
  dispatch({
    type: `${FETCH_DATASTORES}`,
    payload: API.get(url)
  });

export const fetchDatastoresAction = url => {
  const uri = new URI(url);
  return _getDatastoresActionCreator(uri.toString());
};

const _getNetworksActionCreator = url => dispatch =>
  dispatch({
    type: `${FETCH_NETWORKS}`,
    payload: API.get(url)
  });

export const fetchNetworksAction = url => {
  const uri = new URI(url);
  return _getNetworksActionCreator(uri.toString());
};

export const setMigrationsFilterAction = filter => ({
  type: V2V_SET_MIGRATIONS_FILTER,
  payload: filter
});

export const retryMigrationAction = planId => ({
  type: V2V_RETRY_MIGRATION,
  payload: planId
});

export const showDeleteConfirmationModalAction = () => ({
  type: SHOW_DELETE_CONFIRMATION_MODAL,
  payload: true
});

export const hideDeleteConfirmationModalAction = () => ({
  type: HIDE_DELETE_CONFIRMATION_MODAL,
  payload: false
});

export const setMappingToDeleteAction = mapping => dispatch => {
  dispatch({
    type: SET_MAPPING_TO_DELETE,
    payload: mapping
  });
};

export const yesToDeleteInfrastructureMappingAction = () => dispatch => {
  dispatch({
    type: YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL
  });
};

export const deleteInfrastructureMappingAction = mapping => dispatch =>
  dispatch({
    type: DELETE_INFRASTRUCTURE_MAPPING,
    payload: new Promise((resolve, reject) => {
      API.post(`/api/transformation_mappings/${mapping.id}`, {
        action: 'delete'
      })
        .then(response => {
          resolve(response);
        })
        .catch(e => reject(e));
    })
  });

const _archiveTransformationPlanActionCreator = url => dispatch =>
  dispatch({
    type: ARCHIVE_TRANSFORMATION_PLAN,
    payload: API.post(url, { action: 'archive' })
  });

export const archiveTransformationPlanAction = (url, id) => {
  const uri = new URI(`${url}/${id}`);
  return _archiveTransformationPlanActionCreator(uri.toString());
};

const _deleteTransformationPlanActionCreator = url => dispatch =>
  dispatch({
    type: DELETE_TRANSFORMATION_PLAN,
    payload: API.post(url, { action: 'delete' })
  });

export const deleteTransformationPlanAction = (url, id) => {
  const uri = new URI(`${url}/${id}`);
  return _deleteTransformationPlanActionCreator(uri.toString());
};

export const toggleScheduleMigrationModal = plan => ({
  type: V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL,
  payload: plan
});

export const scheduleMigration = payload => dispatch =>
  dispatch({
    type: V2V_SCHEDULE_MIGRATION,
    payload: new Promise((resolve, reject) => {
      const {
        scheduleTime,
        plan: { id: planId }
      } = payload;
      const scheduleId = (payload.plan.schedules && payload.plan.schedules[0].id) || null;
      let url = `/api/service_templates/${planId}`;
      let body = {
        action: 'order',
        resource: {
          schedule_time: scheduleTime
        }
      };
      if (scheduleId) {
        url = `${url}/schedules/${scheduleId}`;
        body = scheduleTime
          ? {
              action: 'edit',
              resource: {
                run_at: { ...payload.plan.schedules[0].run_at, start_time: new Date(scheduleTime).toUTCString() }
              }
            }
          : { action: 'delete' };
      }
      return API.post(url, body)
        .then(response => {
          resolve(response);
          let msg = __('Migration successfully unscheduled');
          if (scheduleTime) {
            msg = sprintf(
              __('Migration successfully scheduled for %s'),
              formatDateTime(response.data.run_at.start_time)
            );
          }
          dispatch({
            type: V2V_NOTIFICATION_ADD,
            message: msg,
            notificationType: 'success',
            actionEnabled: false
          });
        })
        .catch(e => reject(e));
    })
  });
