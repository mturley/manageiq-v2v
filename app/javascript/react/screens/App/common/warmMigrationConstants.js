export const PRECOPYING_DISKS = 'precopying_disks';
