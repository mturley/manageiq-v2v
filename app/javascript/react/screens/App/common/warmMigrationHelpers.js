import { PRECOPYING_DISKS } from './warmMigrationConstants';

// Ignore copy objects that do not include progress data
const filterCopies = copies => copies.filter(copy => copy.copied && copy.to_copy && copy.start_time);

// Find Math.min or Math.max among actual number values, excluding undefined/null.
const getNumericExtreme = (values, mathFn) => {
  const validNumbers = values.filter(value => typeof value === 'number');
  if (validNumbers.length === 0) return null;
  return mathFn(...validNumbers);
};

// Calculate size totals and timestamp extremes from a set of copies
// TODO when statuses are added to copies, also include the "worst" available status
const reduceCopies = copies =>
  copies.reduce(
    (prev, copy) => ({
      totalCopied: prev.totalCopied + (copy.copied || 0),
      totalToCopy: prev.totalToCopy + (copy.to_copy || 0),
      earliestStartTime: getNumericExtreme([prev.earliestStartTime, copy.start_time], Math.min),
      latestEndTime: getNumericExtreme([prev.latestEndTime, copy.end_time], Math.max)
    }),
    {
      totalCopied: 0,
      totalToCopy: 0,
      earliestStartTime: null,
      latestEndTime: null
    }
  );

export const reduceCopiesFromTask = task => {
  const disks = task.options.virtv2v_disks;
  if (!disks || disks.length === 0) return [];
  const allCopiesByDisk = disks.map(disk => (disk.copies ? filterCopies(disk.copies) : [])); // [diskIndex][copyIndex]
  const numCopies = Math.max(...allCopiesByDisk.map(copies => copies.length));
  const reducedCopies = [...new Array(numCopies)].map((_, copyIndex) => {
    const perDisk = allCopiesByDisk.map(diskCopies => diskCopies[copyIndex] || {});
    return { perDisk, ...reduceCopies(perDisk) };
  });
  return reducedCopies; // One array of copies, reduced from the Nth copy of each disk
};

export const getPlanCopySummary = planRequestWithTasks => {
  const tasks = planRequestWithTasks.miq_request_tasks;
  if (!tasks || tasks.length === 0) return {};
  const isPreCopyingAllVms = tasks.every(
    task => task.options && task.options.progress && task.options.progress.current_state === PRECOPYING_DISKS
  );
  const allCopiesByTask = tasks.map(task => reduceCopiesFromTask(task));
  const hasInitialCopyFinished =
    allCopiesByTask.length > 0 &&
    allCopiesByTask.every(reducedCopies => reducedCopies.some(copy => !!copy.latestEndTime));
  return {
    isPreCopyingAllVms,
    allCopiesByTask,
    hasInitialCopyFinished
  };
};
