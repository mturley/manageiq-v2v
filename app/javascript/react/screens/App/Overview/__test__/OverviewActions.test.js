import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';

import { mockRequest, mockReset } from '../../../../../common/mockRequests';
import * as actions from '../OverviewActions';
import { requestTransformationPlansData } from '../overview.transformationPlans.fixtures';
import { cancelRequestResponse } from '../overview.cancelRequest.fixtures';
import {
  SHOW_CONFIRM_MODAL,
  HIDE_CONFIRM_MODAL,
  SHOW_PLAN_WIZARD,
  SHOW_PLAN_WIZARD_EDIT_MODE,
  SHOW_EDIT_PLAN_TITLE_MODAL,
  HIDE_EDIT_PLAN_TITLE_MODAL,
  FETCH_PROVIDERS,
  FETCH_PROVIDERS_URL,
  TRANSFORMATION_PLAN_REQUESTS_URL
} from '../OverviewConstants';

const middlewares = [thunk, promiseMiddleware()];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('confirm modal actions', () => {
  test('show', () => {
    const modalOptions = { mock: 'data' };
    const action = actions.showConfirmModalAction(modalOptions);
    expect(action).toEqual({
      type: SHOW_CONFIRM_MODAL,
      payload: modalOptions
    });
  });

  test('hide', () => {
    expect(actions.hideConfirmModalAction()).toEqual({
      type: HIDE_CONFIRM_MODAL
    });
  });
});

describe('plan wizard actions', () => {
  test('show in normal mode', () => {
    const id = '12345';
    store.dispatch(actions.showPlanWizardAction(id));
    expect(store.getActions()).toEqual([
      {
        type: SHOW_PLAN_WIZARD,
        payload: { id }
      }
    ]);
  });

  test('show in edit mode', () => {
    const id = '12345';
    store.dispatch(actions.showPlanWizardEditModeAction(id));
    expect(store.getActions()).toEqual([
      {
        type: SHOW_PLAN_WIZARD_EDIT_MODE,
        editingPlanId: id
      }
    ]);
  });
});

describe('edit plan title modal', () => {
  test('show', () => {
    const id = '12345';
    store.dispatch(actions.showEditPlanNameModalAction(id));
    expect(store.getActions()).toEqual([
      {
        type: SHOW_EDIT_PLAN_TITLE_MODAL,
        editingPlanId: id
      }
    ]);
  });

  test('hide', () => {
    store.dispatch(actions.hideEditPlanNameModalAction());
    expect(store.getActions()).toEqual([
      {
        type: HIDE_EDIT_PLAN_TITLE_MODAL
      }
    ]);
  });
});

describe('fetchProvidersAction', () => {
  test('dispatches PENDING and FULFILLED actions', () => {
    const response = { data: { mock: 'data' } };
    mockRequest({
      method: 'GET',
      url: FETCH_PROVIDERS_URL,
      status: 200,
      response
    });
    return store.dispatch(actions.fetchProvidersAction()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    const response = 'error';
    mockRequest({
      method: 'GET',
      url: FETCH_PROVIDERS_URL,
      status: 404
    });
    return store.dispatch(actions.fetchProvidersAction()).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});

describe('fetchTransformationPlansAction', () => {
  const { fetchTransformationPlansUrl, response } = requestTransformationPlansData;

  test('dispatches PENDING and FULFILLED actions', () => {
    mockRequest({
      method: 'GET',
      url: fetchTransformationPlansUrl,
      status: 200,
      response
    });

    return store
      .dispatch(
        actions.fetchTransformationPlansAction({
          url: fetchTransformationPlansUrl,
          archived: false
        })
      )
      .then(() => {
        const actions = store.getActions();
        expect(actions).toHaveLength(3);
        expect(actions[0].type).toBe('FETCH_V2V_TRANSFORMATION_PLANS_PENDING');
        expect(actions[1].type).toBe('FETCH_V2V_ALL_REQUESTS_WITH_TASKS_PENDING');
        expect(actions[2].type).toBe('FETCH_V2V_TRANSFORMATION_PLANS_FULFILLED');
        expect(actions[2].payload.data.resources).toHaveLength(8);
      });
  });

  test('dispatches PENDING and REJECTED actions', () => {
    mockRequest({
      method: 'GET',
      url: fetchTransformationPlansUrl,
      status: 404,
      response
    });

    return store
      .dispatch(
        actions.fetchTransformationPlansAction({
          url: fetchTransformationPlansUrl,
          archived: false
        })
      )
      .catch(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});

describe('setMigrationsFilterAction', () => {
  const activeFilter = 'Migrations Plans Not Started';

  test('dispatches an action with correct type and payload', () => {
    store.dispatch(actions.setMigrationsFilterAction(activeFilter));

    expect(store.getActions()).toMatchSnapshot();
  });

  test('dispatches additional actions', () => {
    const SOME_ACTION_TYPE = 'SOME_ACTION_TYPE';
    const payload = 'Some payload';
    store.dispatch(actions.setMigrationsFilterAction(activeFilter, { [SOME_ACTION_TYPE]: payload }));

    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('cancelPlanRequestAction', () => {
  const request = {
    method: 'POST',
    url: TRANSFORMATION_PLAN_REQUESTS_URL,
    data: { action: 'cancel' }
  };

  const action = actions.cancelPlanRequestAction(TRANSFORMATION_PLAN_REQUESTS_URL, '1');

  test('dispatches PENDING and FULFILLED actions', () => {
    mockRequest({
      ...request,
      status: 200,
      response: { data: cancelRequestResponse }
    });

    return store.dispatch(action).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  test('dispatched PENDING and REJECTED actions', () => {
    mockRequest({
      ...request,
      status: 404
    });

    return store.dispatch(action).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
