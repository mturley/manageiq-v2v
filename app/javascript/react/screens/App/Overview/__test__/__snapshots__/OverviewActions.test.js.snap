// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cancelPlanRequestAction dispatched PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "payload": "/api/requests/1",
    "type": "V2V_CANCEL_PLAN_REQUEST_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "V2V_CANCEL_PLAN_REQUEST_REJECTED",
  },
]
`;

exports[`cancelPlanRequestAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "payload": "/api/requests/1",
    "type": "V2V_CANCEL_PLAN_REQUEST_PENDING",
  },
  Object {
    "payload": Object {
      "data": Object {
        "href": "http://0.0.0.0:8080/api/requests/28",
        "message": "MiqRequest 28 canceled",
        "success": true,
      },
    },
    "type": "V2V_CANCEL_PLAN_REQUEST_FULFILLED",
  },
]
`;

exports[`fetchProvidersAction dispatches PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "FETCH_PROVIDERS_PENDING",
  },
  Object {
    "payload": Object {
      "data": Object {
        "mock": "data",
      },
    },
    "type": "FETCH_PROVIDERS_FULFILLED",
  },
]
`;

exports[`fetchProvidersAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_PROVIDERS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_PROVIDERS_REJECTED",
  },
]
`;

exports[`fetchTransformationPlansAction dispatches PENDING and REJECTED actions 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_V2V_TRANSFORMATION_PLANS_REJECTED",
  },
]
`;

exports[`setMigrationsFilterAction dispatches additional actions 1`] = `
Array [
  Object {
    "payload": "Migrations Plans Not Started",
    "type": "V2V_SET_MIGRATIONS_FILTER",
  },
  Object {
    "payload": "Some payload",
    "type": "SOME_ACTION_TYPE",
  },
]
`;

exports[`setMigrationsFilterAction dispatches an action with correct type and payload 1`] = `
Array [
  Object {
    "payload": "Migrations Plans Not Started",
    "type": "V2V_SET_MIGRATIONS_FILTER",
  },
]
`;
